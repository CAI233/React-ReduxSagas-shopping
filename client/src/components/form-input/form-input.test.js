import React from 'react';
import { shallow } from 'enzyme';
import FormInput from './form-input.component';

describe(" 测试表单 ",()=>{
    let wrapper,mockProps;
    beforeEach(()=>{
        mockProps = {
            label: 'test',
            handleChange: jest.fn(),
            email: 'test@gmail.com',
            value: 123
        }
        wrapper = shallow( <FormInput { ...mockProps } /> );
    });
    it("FormInputSnapshot",()=>{
        expect(wrapper).toMatchSnapshot();
    });
    it("测试表单输入",()=>{
        wrapper.find("FromInputStyled").simulate('change');         // simulate('change'): 表单模拟输入( 等待笔记 )
        expect( mockProps.handleChange ).toHaveBeenCalled();
    });
    it("测试由label决定的渲染组件",()=>{
        expect( !!mockProps.label ).toEqual( wrapper.exists("FormInputLabel") );
    })
});