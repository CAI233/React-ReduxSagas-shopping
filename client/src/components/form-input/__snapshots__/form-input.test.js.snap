// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 测试主页目录显示  FormInput Snapshot 1`] = `undefined`;

exports[` 测试主页目录显示  FormInputSnapshot 1`] = `ShallowWrapper {}`;

exports[` 测试表单  FormInputSnapshot 1`] = `ShallowWrapper {}`;
